File: app/src/main/java/org/deiverbum/app/data/entity/LHPrayerAll.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.model.Oracion;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class LHPrayerAll {
    @Embedded
    public LHPrayerEntity lhPrayerEntity;

    @Relation(
            parentColumn = "prayerFK",
            entityColumn = "prayerID",
            entity = PrayerEntity.class
    )
    public PrayerEntity prayerEntity;

    public Oracion getDomainModel() {
        Oracion theModel=new Oracion();
        theModel.setTexto(prayerEntity.getTexto());
        return theModel;
    }
}

File: app/src/main/java/org/deiverbum/app/core/database/model/relation/LHPrayerAll.kt
package org.deiverbum.app.core.database.model.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.core.database.model.entity.LHPrayerEntity
import org.deiverbum.app.core.database.model.entity.PrayerEntity
import org.deiverbum.app.core.database.model.entity.asExternalModel

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class LHPrayerAll(
    @Embedded
    var lhPrayerEntity: LHPrayerEntity,

    @Relation(parentColumn = "prayerFK", entityColumn = "prayerID", entity = PrayerEntity::class)
    var prayerEntity: PrayerEntity
)

fun LHPrayerAll.asExternalModel() = prayerEntity.asExternalModel()

